docs/kratos/self-service/flows/verify-email-account-activation.mdx

```diff
@@ -183,408 +134,141 @@ your Identity Schema:
-    "required": true,
-    "disabled": false
-  },
-  "messages": null,
-  "meta": {}
-}
-{
-  "type": "input",
-  "group": "link",
-  "attributes": {
-    "name": "method",
-    "type": "submit",
-    "value": "link",
-    "disabled": false
-  },
-  "messages": null,
-  "meta": {
-    "label": {
-      "id": 1070005,
-      "text": "Submit",
-      "type": "info"
-    }
-  }
-}
-```
+   ```diff
+   selfservice:
+     flows:
+       login:
+   +      after:
+   +        hooks:
+   +        - hook: require_verified_address
+   ```
 
-## Verification form validation
+3. Update the Ory Identities configuration using the file you worked with:
 
-The form payloads are then submitted to Ory Identities which follows up with:
+   ```shell
+   ory update identity-config {project-id} --file updated_config.yaml
+   ```
 
-- An HTTP 303 See Other redirect pointing to the Registration UI for Browser Clients;
-- An `application/json` response for API Clients and Client-Side Browser applications (for example Single Page Apps)e Apps).
+:::tip
 
-### Verification Link via Email
+For more information see the [hooks configuration](../../hooks/01_configure-hooks.mdx) documentation.
 
-To send the verification email, the end-user fills out the form. There might be validation errors such as a malformed email:
+:::
 
-<CodeTabs items={getFlowMethodLinkWithErrors} />
+```mdx-code-block
+  </TabItem>
+</Tabs>
+```
 
-When validation errors happen, browser clients receive a HTTP 303 See Other redirect to the Verification Flow UI, containing the
-Verification Flow ID which includes the error payloads.
+## Choosing the right strategy
 
-For API Clients, the server typically responds with HTTP 400 Bad Request and the Verification Flow in the response payload as
-JSON.
+Ory supports two strategies for verifying your user's addresses.
 
-#### Successful submission
+```mdx-code-block
+<Tabs>
+  <TabItem value="code" label="One-time codes">
+```
 
-On successful submission, an email will be sent to the provided address:
+When using one-time codes to verify a user's address, Ory Identities sends out an email containing a 6-digit code. The user must
+enter the code in a dedicated UI text field to verify their address.
 
-<CodeTabs items={getFlowMethodLinkSuccess} />
+Alternatively, the email also contains a link to the form that pre-fills the code if the user clicks on it. The user must still
+submit the form for Ory Identities to actually mark the address as verified.
 
-## Unsuccessful verification
+```mdx-code-block
+  </TabItem>
+  <TabItem value="link" label="Magic Links">
+```
 
-If the verification challenge (for example the link in the verification email) is invalid or expired, the user will be HTTP 303
-redirected to the Verification UI.
+When using the "magic link" method, Ory Identities sends the user an email with a "magic link". If the user has access to the
+email associated with the and clicks the link, Ory Identities marks that address as verified.
```

Comment:
```suggestion
When using the "magic link" method, Ory Identities sends the user an email with a link that serves as the means of verification. If the user has access to the
email associated with the account and clicks the link, Ory Identities marks that address as verified.
```

Reply:
Not really sure if that sentence ("that serves as the means of verification") adds anything more, TBH. Would you be okay with leaving it out?

Reply:
"that serves as the means of verification" is an explanation of what is that "magic link" that we refer to. By reading that the user knows that the "magic link" is a link sent for the purpose of verification. Saying that the "magic link methods sens magic links in emails" is too vague. What is the link for in that case? 
But maybe it'd work better if it were shorter? If you have an idea how to do that then sure, please show me :)

Reply:
Yeah, but that's explained in the sentence afterwards, no?

Reply:
When using the "magic link" method, Ory Identities sends the user an email with a **verification** link. If the user has access to the
email associated with the account and clicks the link, Ory Identities marks that address as verified.

What about this? (bold for highlighting)

Reply:
That sounds good! Go ahead and use it. No bolding, of course ;)
https://github.com/ory/docs#bolding-code-formatting